### PR TITLE
[REVIEW] The to_host logic for null strings should not modify input array

### LINKED ISSUE #330: NVStrings::to_host may overwrite input pointers
The input to `to_host()` is a an array of pointers to char arrays. This line, if called, will set the pointers in the array to all zeros. Since these pointers point to arrays allocated outside of NVStrings, this might be surprising to the caller, and lead to lost memory. 

https://github.com/rapidsai/custrings/blob/fd4d072fc518de400c88890069c8739183e2563f/cpp/src/NVStrings.cu#L291

This line will have the same ill effect:
https://github.com/rapidsai/custrings/blob/fd4d072fc518de400c88890069c8739183e2563f/cpp/src/NVStrings.cu#L349

The documentation doesn't mention these side effects, so I assume they are bugs.

### PR DESCRIPTION
**Pull request by davidwendt**
closes #330

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 - PR #306 Set main label always for libnvstrings
 - PR #307 Update conda dependencies
 - PR #326 Update python doc version
+- PR #335 The to_host logic for null strings should not modify input array
```
Comment (harrism) [inline CHANGELOG.md]: Say what the change is, not what it should do.
```suggestion
- PR #335 to_host no longer overwrites input pointers
```

File: cpp/src/NVStrings.cu [hunk 2]
```diff
@@ -345,17 +334,10 @@ int NVStrings::to_host(char** list, int start, int end)
     for( unsigned int idx=0; idx < count; ++idx )
     {
         if( h_strings[idx]==0 )
```
Comment (harrism) [inline cpp/src/NVStrings.cu]: `continue` leads to spaghetti logic. It's clearer to write:

```
for( unsigned int idx=0; idx < count; ++idx )
{
  if( h_strings[idx]!=0 ) // Skip null strings
  {
    size_t offset = h_offsets[idx];
    size_t len = h_offsets[idx+1] - offset;
    const char* p_data = h_buffer + offset;
    if( list[idx] )
      memcpy(list[idx], p_data, len-1);
  }
}
```

Comment (davidwendt) [reply cpp/src/NVStrings.cu]: I have to disagree with this. I think using continue for exception cases at the top of a loop is better than adding extra indenting for the main code logic for the loop's purpose.

**Comment by jrhemstad** [comment]
Was there a test for this behavior that could be added to make sure it's fixed and doesn't reoccur?

**Pull request merged**